be/src/util/dynamic_cache.h

```diff
@@ -331,6 +314,37 @@ class DynamicCache {
         return _size <= _capacity;
     }
 
+    // clear all unused *and* expired objects, but make sure the time cost must less than
+    // `timeout_millis`, to avoid hold `_lock` too long.
+    // if `timeout_millis` <= 0, that means we can ignore it.
+    void _clear_expired(int64_t timeout_millis) {
```

Comment:
Why not move it another data structure, and not conflict with the dynamic cache?